pkg/pillar/pubsub/pubsub.go

```diff
@@ -72,26 +72,26 @@ const (
 	PersistConfigDir = PersistDir + "/config"
 )
 
-func Publish(agentName string, topicType interface{}) (*Publication, error) {
+func Publish(agentName string, topicType interface{}) (*PublicationImpl, error) {
```

Comment:
I missed this in my review. This is a problem. It doesn't break anything now, but it makes the interface useless.

Here, we return `*PublicationImpl`, rather than `Publication`. This means that consuming code - anything that calls `pubsub.Publish()` - is getting back a `PublicationImpl` (specific struct) rather than `Publication` (an interface), and this is just as tied to it as before.

These should return a `Publication`. We need a new PR for this.

Reply:
Yeah.. Was thinking about it.. Right now, the unit tests cannot use Publish() to create the publisher. The tests create the publisher separately.. The part of the code where we create pubsub need to be bypassed for unit tests - as the tests create their own pubsubs..